spring-cloud-stream-binder-rabbit/src/test/java/org/springframework/cloud/stream/binder/rabbit/RabbitTestBinder.java

```diff
@@ -65,6 +66,9 @@ public RabbitTestBinder(ConnectionFactory connectionFactory, RabbitMessageChanne
 		scheduler.setPoolSize(1);
 		scheduler.afterPropertiesSet();
 		context.getBeanFactory().registerSingleton(IntegrationContextUtils.TASK_SCHEDULER_BEAN_NAME, scheduler);
+		PublishSubscribeChannel errorChannel = new PublishSubscribeChannel();
+		context.getBeanFactory().initializeBean(errorChannel, IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME);
+		context.getBeanFactory().registerSingleton(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME, errorChannel);
```

Comment:
Can't we use something more closer to the real Integration infrastructure:
```
AnnotationConfigApplicationContext context = new AnnotationConfigApplicationContext();
context.register(VanillaIntegration.class);
...
@EnableIntegration
private static class VanillaIntegration {

}
```
instead ?